### PR TITLE
Feature/3 improve logging

### LINKED ISSUE #3: Improve logging
It should always be logged on INFO level that an additional remote repository has been added (https://github.com/Netcentric/maven-ext-repos-from-env/blob/c19a28bfe36bbc6508986fdc8d2cfb8dc0c8bc89/src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java#L74)

In addition some more debug logs would be helpful which expose

1.  that not all necessary environment variables are set (which ones?)
1. which remote repositories are already considered (https://maven.apache.org/ref/3.3.9/maven-core/apidocs/org/apache/maven/execution/MavenExecutionRequest.html#getRemoteRepositories()) to be able to detect duplicate remote repo URLs
1. which mirrors are currently configured (https://maven.apache.org/ref/3.3.9/maven-core/apidocs/org/apache/maven/execution/MavenExecutionRequest.html#getMirrors()) as mirror may prevent the feature from working IMHO

### PR DESCRIPTION
**Pull request by ghenzler**
Closes #3

### CODE REVIEW

**Comment by ghenzler** [comment]
#5 should be merged first (to only have one commit on this PR)

File: src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java [hunk 1]
```diff
@@ -41,47 +43,80 @@
...
+            logger.debug(msg);
+        }
+    }
+
+    private void logRepositoriesAndMirrors(MavenExecutionRequest request) {
+
+        List<ArtifactRepository> repositories = request.getRemoteRepositories();
+        List<ArtifactRepository> pluginRepositories = request.getPluginArtifactRepositories();
+        List<Mirror> mirrors = request.getMirrors();
+        logVerbose("Configured in settings.xml: repositores: "+
```
Comment (kwin) [inline src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java]: Not necessarily only in settings.xml, right? Doesn't this also expose the repos from pom.xml?

File: src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java [hunk 2]
```diff
@@ -41,47 +43,80 @@
...
             // activate profile
             repositoriesFromEnv.setId(PROFILE_ID_REPOSITORIES_FROM_ENV);
             request.addProfile(repositoriesFromEnv);
             request.addActiveProfile(PROFILE_ID_REPOSITORIES_FROM_ENV);
+            
         }
     }
 
+    private void logVerbose(String msg) {
+        if(isVerbose) {
```
Comment (kwin) [inline src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java]: formatter seems not correctly set up, either space between if and "(" everywhere or not

File: src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java [hunk 3]
```diff
@@ -41,47 +43,80 @@
...
+        }
+    }
+
+    private void logRepositoriesAndMirrors(MavenExecutionRequest request) {
+
+        List<ArtifactRepository> repositories = request.getRemoteRepositories();
+        List<ArtifactRepository> pluginRepositories = request.getPluginArtifactRepositories();
+        List<Mirror> mirrors = request.getMirrors();
+        logVerbose("Configured in settings.xml: repositores: "+
+                repositories.stream().map(ArtifactRepository::getId).collect(Collectors.joining(","))
```
Comment (kwin) [inline src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java]: even more interesting is the URL

Comment (kwin) [reply src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java]: and snapshot policy

**Review by kwin — CHANGES_REQUESTED** [review]

File: src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java [hunk 4]
```diff
@@ -41,47 +43,80 @@
...
 
+    private boolean isVerbose;
+    
     @Override
     public void process(CliRequest cliRequest) throws Exception {
-
-        List<RepoFromEnv> reposFromEnv = getReposFromEnv(System.getenv());
+        
+        Map<String, String> sysEnv = System.getenv();
+        isVerbose = Boolean.valueOf(sysEnv.get(ENV_PROP_MVN_SETTINGS_REPO_VERBOSE));
```
Comment (jon-entwistle) [inline src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java:64]: Why is this needed? Is the verbosity level not picked up automatically from the -X parameter?

File: src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java (hunk 2 above)
Comment (jon-entwistle) [inline src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java]: This looks really strange - why is the log not left at debug, which is printed (at level debug) if that level of verbosity is activated?

File: src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java (hunk 1 above)
Comment (ghenzler) [reply src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java]: At this time of the lifecycle you only get a hold of what is configured in settings.xml

File: src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java (hunk 2 above)
Comment (kwin) [reply src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java]: debug (-X) emits a lot of output. The flag allows to only emit logs relevant for this feature!
The other alternative is to tweak simplelogger.properties (https://maven.apache.org/maven-logging.html#configuring-logging)

File: src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java (hunk 4 above)
Comment (ghenzler) [reply src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java:64]: It is, but if you run a typical build with `-X` you get a log file which is 50x larger than normal - allowing to set this also via system property allows to specifically get what you want (while the regular log "stays normal")

File: src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java (hunk 3 above)
Comment (ghenzler) [reply src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java]: adding this as well then

File: src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java (hunk 4 above)
Comment (jon-entwistle) [reply src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java:64]: OK I would still leverage the log api rather that the if/else - so `if (LOG.isDebugEnabled() || isVerbose)` .... - the output should still appear as debug, not info. Also, this sets the debug on an environmental variable level so it cannot easily be changed - wouldn't it make more sense to use a parameter to maven for one off 'debug' runs, or allow both?

**Review by kwin — APPROVED** [review]

**Review by kwin — CHANGES_REQUESTED** [review]
Still inconsistent formatting with regards to `if`

**Comment by ghenzler** [comment]
@kwin reformatted the whole file

**Review by kwin — APPROVED** [review]
